rust/scx_utils/src/compat.rs

```
@@ -170,17 +170,49 @@ pub fn in_kallsyms(ksym: &str) -> Result<bool> {
     Ok(false)
 }
 
-pub fn cond_kprobe_enable<T>(sym: &str, prog_ptr: &OpenProgramImpl<T>) -> Result<()> {
+pub fn tracepoint_exists(tracepoint: &str) -> Result<bool> {
+    let file = std::fs::File::open("/sys/kernel/debug/tracing/available_events")?;
+    let reader = std::io::BufReader::new(file);
+
+    for line in reader.lines() {
+        for tp in line.unwrap().split_whitespace() {
+            if tracepoint == tp {
+                return Ok(true);
+            }
+        }
+    }
+
+    Ok(false)
+}
+
+pub fn cond_kprobe_enable<T>(sym: &str, prog_ptr: &OpenProgramImpl<T>) -> Result<bool> {
     if in_kallsyms(sym)? {
         unsafe {
             bpf_program__set_autoload(prog_ptr.as_libbpf_object().as_ptr(), true);
         }
+        return Ok(true);
     } else {
         warn!("symbol {} is missing, kprobe not loaded", sym);
     }
-    Ok(())
+
+    Ok(false)
 }
 
+pub fn cond_tracepoint_enable<T>(tracepoint: &str, prog_ptr: &OpenProgramImpl<T>) -> Result<bool> {
```

Since it's a lib, instead of printing the warning directly here, I'd return an error to the caller, like we do with the `tracepoint_exists()` and then change the function to just return a Result.

Something like:
```
pub fn cond_tracepoint_enable<T>(
    tracepoint: &str,
    prog_ptr: &OpenProgramImpl<T>,
) -> Result<()> {
    if tracepoint_exists(tracepoint)? {
        unsafe {
            bpf_program__set_autoload(prog_ptr.as_libbpf_object().as_ptr(), true);
        }
        Ok(())
    } else {
        Err(Error::new(
            ErrorKind::NotFound,
            format!("tracepoint {} is missing, tracepoint not loaded", tracepoint),
        ))
    }
}
```

But this can be improved later, the same applies also to `cond_kprobe_enable()`.

---

That's a good suggestion. After cutting the release, I will submit a separate PR cleaning this up for lavd and layered.